### DISCUSSION TITLE
Crashes aren't being fully captured/clear

### DISCUSSION THREAD

**Discussion by Edstub207** [getsentry/sentry-unreal#109] [Q&A] [answered]
We've finally enabled Sentry using the plugin, in a working condition that the team can utilise. 

However, the crashes are using a non-legacy call stack and the titles don't always seem easily identifiable from a glance (Vs the error message on some crashes, which is a lot clearer) 

![image](https://user-images.githubusercontent.com/13305475/190203690-901c5e2c-51d5-43c5-9966-8131e5bf5f0a.png)

Are there plans to improve this?

We've also had cases where DLLs haven't been uploaded, even though they were using the auto upload functionality, albeit in a different build location (As on a different machine). I understand not wanting to upload the editor itself every compile, but is there a way we can get the most information out of these crash reports? For deployed builds.

**Comment by Edstub207** [comment]
@tustanivsky I wasn't sure if this should be an issue or discussion

**Comment by tustanivsky** [comment, marked as answer]
We're expecting to update the crash capturing mechanism for desktop platforms with UE 5.1 release. In this new version it should be possible to get rid of the embedded Crash Reporter tool and use [Sentry's library](https://github.com/getsentry/sentry-native) instead.

Regarding the DLLs upload only those that are in project/plugin build folders are processed automatically at the moment. However you might want to check the [upload script](https://github.com/getsentry/sentry-unreal/blob/main/plugin-dev/Scripts/upload-debug-symbols-win.ps1) and tweak it accordingly by providing a path to extra libraries.

**Reply by Edstub207** [reply]
@tustanivsky when do you expect the Sentry 5.1 Changes to be available? Would this move the symbols upload to be handled by Sentry itself instead than a Postbuild step? 

We currently have a situation where the symbols we do want uploaded, aren't getting uploaded properly, then the ones we don't are being uploaded. 

So would love more control over what is/isn't uploaded. 

For example, we'd love to basically say "If build is on build machine (usually denoted with the -buildmachine flag) automatically upload" - otherwise a manual upload.

**Reply by tustanivsky** [reply]
The main branch already contains some fixes for UE 5.1 compatibility (#150) and currently we're investigating a [new engine API](https://github.com/EpicGames/UnrealEngine/pull/7976) that should allow us to capture crashes on Windows automatically without using the default Crash Reporter. No changes in terms of debug symbols upload are planned though.
